src/Microsoft.Azure.WebJobs.Extensions.Redis/Bindings/RedisAsyncCollector.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using StackExchange.Redis;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Redis
+{
+    internal class RedisAsyncCollector : IAsyncCollector<string[]>
+    {
+        private readonly IConnectionMultiplexer multiplexer;
+        private readonly string command;
+        private readonly ILogger logger;
+
+        private IBatch batch;
+
+        public RedisAsyncCollector(IConnectionMultiplexer multiplexer, string command, ILogger logger)
+        {
+            this.multiplexer = multiplexer;
+            this.command = command;
+            this.logger = logger;
+        }
+
+        public Task AddAsync(string[] arguments, CancellationToken cancellationToken = default)
+        {
+            if (batch is null)
+            {
+                logger?.LogDebug("Creating batch.");
+                batch = multiplexer.GetDatabase().CreateBatch();
+            }
```

Comment:
Can this be a part of the constructor?

Reply:
Oh I see we're setting in null in the `FlushAsync` method.  Is that the only way to clear out a batch?